inference-engine/src/legacy_api/include/cnn_network_impl.hpp

```diff
@@ -30,6 +31,9 @@ namespace details {
 class INFERENCE_ENGINE_API_CLASS(CNNNetworkImpl): public ICNNNetwork {
 public:
     CNNNetworkImpl();
+    explicit CNNNetworkImpl(const ICNNNetwork & ngraphImpl,
+        const ngraph::pass::PassParam::param_callback & callback = ngraph::pass::PassParam::getDefaultCallback(),
```

Comment:
Why do we need to keep this parametrization callback if it's only needed for on demand conversion where callback is not used?

Reply:
it's used in CPU & GPU as well where we have a callback.

Reply:
Removed new ctor usage for cases of custom transformation pipeline as discussed with @GlebKazantaev offline.